helper/memory_calculator_test.go

```diff
@@ -361,6 +363,25 @@ func testMemoryCalculator(t *testing.T, context spec.G, it spec.S) {
 				})
 			})
 
+			context("$JAVA_TOOL_OPTIONS with agents", func() {
+				it.Before(func() {
+					Expect(os.Setenv("JAVA_TOOL_OPTIONS", fmt.Sprintf("-javaagent:%s", filepath.Join("../count/testdata", "stub-dependency.jar")))).To(Succeed())
+				})
+
+				it.After(func() {
+					Expect(os.Unsetenv("JAVA_TOOL_OPTIONS")).To(Succeed())
+				})
+
+				it("counts classes of agent jars supplied via $JAVA_TOOL_OPTIONS", func() {
+					c, err := helper.CountAgentClasses(os.Getenv("JAVA_TOOL_OPTIONS"))
+					Expect(err).NotTo(HaveOccurred())
+					Expect(c).To(Equal(2))
+					Expect(m.Execute()).To(Equal(map[string]string{
+						"JAVA_TOOL_OPTIONS": fmt.Sprintf("-javaagent:%s -XX:MaxDirectMemorySize=10M -Xmx522705K -XX:MaxMetaspaceSize=13870K -XX:ReservedCodeCacheSize=240M -Xss1M", filepath.Join("../count/testdata", "stub-dependency.jar")),
+					}))
+				})
+			})
+
```

Comment:
A couple of test cases came to mind. Do you think we have these test cases covered?

1. It errors if `-javaagent` is garbage or the jar doesn't exist. Part of me wonders if we should actually error in this case, or just warn like "Couldn't not read <foo>, class count and metaspace may not be sized correctly".
2. It works if `-javaagent` is not set.
3. If a user has fixed metaspace, we are not changing it (or possibly we skip the whole process).

Reply:
1. Covered - now only checking the agent file itself - skipping & warning if file-not-found
2. Covered - if no agents, the count is 0 and the calculation proceeds
3. The process happens before calculation/determination of flag provenance, so covered by checking user Metaspace doesn't get changed